src/subscriber.rs

```diff
@@ -125,6 +125,44 @@ impl Drop for Subscriber {
 }
 
 impl Subscriber {
+    /// Creates a future that resolves to the next value of the
+    /// subscriber, or None if the backing `Db` shuts down
+    pub fn next(&mut self) -> impl Future<Output = Option<Event>> + '_ {
```

Comment:
Although this is returning a concrete Next<'_> type in the implementation, we don't need to leak the new type in the public API in the event it needs to change later for some reason

Reply:
Realizing now that `next` here conflicts with Iterator's `next` Maybe naming this something else would be better